pkg/tsdb/prometheus/prometheus.go

```diff
@@ -162,13 +163,20 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 		span.SetTag("stop_unixnano", query.End.UnixNano())
 		defer span.Finish()
 
-		value, _, err := client.QueryRange(ctx, query.Expr, timeRange)
+		var results model.Value
 
-		if err != nil {
-			return &result, err
+		switch query.QueryType {
+		case Range:
+			results, _, err = client.QueryRange(ctx, query.Expr, timeRange)
```

Comment:
I am wondering what the second returned var `Warnings` is about. Could it be useful? 🤔

Reply:
Hmmm, I am not sure but that could be very interesting to explore. We currently don't use Warning, but I have added it as the non-MVP point https://github.com/grafana/grafana/issues/37784. Basically to investigate if/how could we use it.